### ISSUE TITLE
[Request for insight] result of layered colors

### ISSUE THREAD

**Issue by haxxxton** [gka/chroma.js#200]
I've been banging my head about this one for a while, and cant quite figure out why the resultant colors between mathematically calculated values (ie, those from chroma-js) and those inside a image editing program (in this case Sketch), are different.

The scenario i find myself is in this:
Our design team has created a color scale from 0% to 200%. 100% being the pure color (eg a blue `#007FEF`), they then overlay, in opacity steps of 10%, a light color (`#FFFFFF`) and a dark color (`#14181C`) moving away from this 100 in opposite directions. Leading to 0% being the pure light color, and 200% being the pure dark color. Below is an example of said spectrum.

![image](https://user-images.githubusercontent.com/743105/49358070-21eb9680-f71d-11e8-88ad-86bbb5a91312.png)

The resultant colors of this overlay technique are determined by using the color picker/dropper tool on the overlayed colors.

When trying to replicate this scale inside chroma, I am trying to make generation of the colors as easy as possible. I assumed that `.mix()` would be best solution for this, however, the resulting hex values are slightly off from the resulting hex colors in my image editing program.

```
/*
mix code				generated output  expected output  matching
*/
chroma.mix('007FEF', 'FFFFFF', 1);	// #FFFFFF	// #FFFFFF	// ===
chroma.mix('007FEF', 'FFFFFF', 0.9);	// #E6F2FD	// #E6F2FD	// ===
chroma.mix('007FEF', 'FFFFFF', 0.8);	// #CCE5FC	// #CCE5FB	// =/=
chroma.mix('007FEF', 'FFFFFF', 0.7);	// #B3D9FA	// #B3D8FA	// =/=
chroma.mix('007FEF', 'FFFFFF', 0.6);	// #99CCF9	// #99CBF8	// =/=
chroma.mix('007FEF', 'FFFFFF', 0.5);	// #80BFF7	// #80BFF7	// ===
chroma.mix('007FEF', 'FFFFFF', 0.4);	// #66B2F5	// #66B2F5	// ===
chroma.mix('007FEF', 'FFFFFF', 0.3);	// #4DA5F4	// #4DA5F3	// =/=
chroma.mix('007FEF', 'FFFFFF', 0.2);	// #3399F2	// #3398F2	// =/=
chroma.mix('007FEF', 'FFFFFF', 0.1);	// #1A8CF1	// #1A8CF0	// =/=
chroma.mix('007FEF', '14181C', 0);	// #007FEF	// #007FEF	// ===
chroma.mix('007FEF', '14181C', 0.1);	// #0275DA	// #0274D9	// =/=
chroma.mix('007FEF', '14181C', 0.2);	// #046AC5	// #046AC5	// ===
chroma.mix('007FEF', '14181C', 0.3);	// #0660B0	// #065FAE	// =/=
chroma.mix('007FEF', '14181C', 0.4);	// #08569B	// #08569A	// =/=
chroma.mix('007FEF', '14181C', 0.5);	// #0A4C86	// #0A4B85	// =/=
chroma.mix('007FEF', '14181C', 0.6);	// #0C4170	// #0C4070	// =/=
chroma.mix('007FEF', '14181C', 0.7);	// #0E375B	// #0E365B	// =/=
chroma.mix('007FEF', '14181C', 0.8);	// #102D46	// #102C45	// =/=
chroma.mix('007FEF', '14181C', 0.9);	// #122231	// #122230	// =/=
chroma.mix('007FEF', '14181C', 1);	// #14181C	// #14181C	// ===
```

Now admittedly, the difference between the two colors is so marginal (<1%) and visually I personally cant tell the difference, I have been tasked with seeing if there is a way to make things match perfectly. There are some base colors where this difference is a little more noticeable (~4% difference) at certain percentages of mixing, this is particularly the case for our bluegray base color `#3D4C5C`. For my own sanity I would prefer not to have to manually type out each hex color (we have ~10 base colors, and some step by 5% instead of 10% opacity).

**TLDR;**
Is there a blending/mixing function in chroma that will mirror exactly the visual output achieved by overlaying a transparent color over an opaque color in Sketch?

**Comment by gka** [comment]
hm, interesting problem! might be just some difference in how chroma.js and Sketch are dealing with rounding..

**Comment by gka** [comment]
let's do the math together with one example color, shall we? 

mixing `#007FEF` with `#FFFFFF` at 80%.

In chroma, `mix` is an alias for `interpolate` and per default, `rgb` interpolation is used. so we end up with this function: https://github.com/gka/chroma.js/blob/master/src/interpolator/interpolate-rgb.coffee

of course before we can interpolate we need the RGB values.
* `#007FEF` is, without doubt, RGB: `[0, 127, 239]` (we can use `parseInt("EF", 16)` to verify)
* `#FFFFFF` is RGB: `[255, 255, 255]`

interpolation means linear interpolation and that means: `R_out = R_1 + f * (R_2 - R_1)` etc. our `f` is 0.8.

* **red:** `0 + 0.8 * (255 - 0)` === 204 
* **green:** `127 + 0.8 * (255 - 127)` === 229.4 
* **blue:** `239 + 0.8 * (255 - 239)` === 251.8

so our result before rounding is `[204, 229.4, 253.4]`. 

now let's compare the two conflicting hex values:

* chroma said the result is: `#CCE5FC` which translates to `[204, 229, 252]`
* Sketch said the result is: `#CCE5FB` which translates to `[204, 229, 251]`

the only difference is the blue component. and as I expected, the difference is due to different rounding strategies. while chroma is rounding `251.8` up to `252`, Sketch is just cutting off the `.8` to `251`.

I'd argue that chroma is "right" and Sketch is wrong, but as you said, the differences are very subtle.

**Comment by gka** [comment]
if you'd want to fork, this is the point you'd want to change: https://github.com/gka/chroma.js/blob/master/src/converter/out/rgb2hex.coffee#L6-L8

I briefly thought about making this an option, but I'm not sure if there are more use cases besides the one you outlined.
